File: annotation-processor-testing/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyDoublePropertyElement.java
package com.tickaroo.tikxml.annotationprocessing.propertyelement;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
public class EmptyDoublePropertyElement {
  @PropertyElement
  double empty;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof EmptyDoublePropertyElement)) return false;

    EmptyDoublePropertyElement that = (EmptyDoublePropertyElement) o;

    return Double.compare(that.empty, empty) == 0;
  }

  @Override public int hashCode() {
    long temp = Double.doubleToLongBits(empty);
    return (int) (temp ^ (temp >>> 32));
  }
}

File: annotationprocessortesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyDoublePropertyElement.kt
package com.tickaroo.tikxml.annotationprocessing.propertyelement

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
class EmptyDoublePropertyElement {
    @PropertyElement
    var empty: Double = 0.toDouble()

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EmptyDoublePropertyElement) return false

        val that = other as EmptyDoublePropertyElement?

        return java.lang.Double.compare(that!!.empty, empty) == 0
    }

    override fun hashCode(): Int {
        val temp = java.lang.Double.doubleToLongBits(empty)
        return (temp xor temp.ushr(32)).toInt()
    }
}
